app/util/ClassToMap.scala

```diff
@@ -0,0 +1,101 @@
+package util
+
+import com.gu.contentatom.thrift.atom.media._
+import com.gu.contentatom.thrift.{ImageAssetDimensions, ImageAsset, Image}
+import shapeless.labelled._
+import shapeless._
+
+object ClassToMap {
+  // FROM CASE CLASS TO MAP (http://stackoverflow.com/a/31638390)
+  trait ToMapRec[L <: HList] {
+    def apply(l: L): Map[String, Any]
+  }
+
+  // Case for heads that don't have ToMapRec instances (hconsToMapRec1).
```

Comment:
Yes, most of the times. Or if there is a special cladd we need to define our own implicit. The case when the head has a ToMapRec is when you can recurse on the head. Shapeless know how to generate representations for case classes and basic types.